src/main/java/seedu/address/model/BorrowerIdGenerator.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model;
+
+import java.util.stream.IntStream;
+
+import seedu.address.model.borrower.BorrowerId;
+
+/**
+ * Borrower ID generator that generates a borrower ID when a new borrower is registered.
+ */
+public class BorrowerIdGenerator {
+    private static final int BORROWER_ID_LENGTH = 4; //excluding prefix 'K'
+    private static final String PREFIX = "K";
+
+    private static int currentBorrowerIdIndex = 0;
+
+    private static BorrowerRecords borrowers;
+
+    public static void setBorrowers(BorrowerRecords borrowers) {
+        BorrowerIdGenerator.borrowers = borrowers;
+        currentBorrowerIdIndex = 0;
+    }
+    /**
+     * Generates a new borrower ID based on the current borrower ID index.
+     */
+    public static BorrowerId generateBorrowerId() {
+        currentBorrowerIdIndex++;
+        int paddingLength = getPaddingLength();
+        String padding = getPadding(paddingLength);
+        BorrowerId id = new BorrowerId(PREFIX + padding + currentBorrowerIdIndex);
+        while (borrowers.checkIfBorrowerIdExists(id)) {
+            currentBorrowerIdIndex++;
```

Comment:
You should update id here to ensure that the while loop does not do on infinitely.